common/scala/src/main/scala/whisk/common/Logging.scala

```diff
@@ -178,6 +179,34 @@ object LogMarkerToken {
   }
 }
 
+trait WhiskMetric {}
+
+case class WhiskCounterMetric(component: String, action: String) extends WhiskMetric {}
+
+case class WhiskHistogramMetric(component: String, action: String, value: Long) extends WhiskMetric
+
+object MetricEmitter {
+
+  def getMetricName(component: String, action: String, isFailed: Boolean = false): String = {
+    if (isFailed) {
+      component + "_" + action + "_failed"
+    } else {
+      component + "_" + action
+    }
+  }
+
+  def emitMetric(wm: WhiskMetric, isFailed: Boolean = false) = wm match {
+    case WhiskCounterMetric(component, action) =>
+      Kamon.metrics
+        .counter(getMetricName(component, action, isFailed))
+        .increment(1)
+    case WhiskHistogramMetric(component, action, value) =>
+      Kamon.metrics
+        .histogram(getMetricName(component, action, isFailed))
+        .record(value)
+  }
```

Comment:
Why not have an overloaded version of `emitMetric` and get rid of the intermediary types + non-needed allocations?

Reply:
I have simplified the MetricsEmitter (via amend) to reduce allocations